### PR TITLE
Correct issues raised by shellcheck

### PR DESCRIPTION
**Pull request by alexwlchan**
Supersedes #578 by actually fixing the issues raised by shellcheck. Of course this may have broken something, because shell script hilarity. All the proposed fixes _look_ sensible, but I’d like the CI to have a crack at it first.

### CODE REVIEW

**Comment by alexwlchan** [comment]
@DRMacIver I approved your commits up to the end of #578 – mind casting your eye over my one commit to fix the issues raised by shellcheck?

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,8 @@ sudo: false
 
 env: PYTHONDONTWRITEBYTECODE=x
 
+ghc: 7.8
```
Comment (DRMacIver) [inline .travis.yml]: I think I was the one who added this, but it's actually not needed - we ended up installing GHC ourself (because that's how this sort of thing inevitably goes in Hypothesis build land. Sigh) using stack.

File: scripts/check-ancient-pip.sh [hunk 2]
```diff
@@ -7,19 +7,19 @@ PYTHON=$1
 
 BROKEN_VIRTUALENV=$($PYTHON -c'import tempfile; print(tempfile.mkdtemp())')
 
-trap "rm -rf $BROKEN_VIRTUALENV" EXIT
+trap 'rm -rf $BROKEN_VIRTUALENV' EXIT
```
Comment (DRMacIver) [inline scripts/check-ancient-pip.sh]: It's not totally obvious to me that this is doing the right thing, because I don't understand the semantics of variable evaluation in trap. Previously this would have evaluated the BROKEN_VIRTUALENV variable at point of definition. Does it evaluate it at point of execution now and is that variable in scope there?

File: scripts/install.sh [hunk 3]
```diff
@@ -23,43 +23,43 @@ env | grep UTF
...
 
-  if (( $(date '+%s') > 300 + $(stat --format=%X $LOCKFILE) )); then
+  if (( $(date '+%s') > 300 + $(stat --format=%X "$LOCKFILE") )); then
     echo "We've waited long enough"
-    rm -rf $LOCKFILE
+    rm -rf "$LOCKFILE"
   fi
 done
-trap "rm -rf $LOCKFILE" EXIT
+trap 'rm -rf $LOCKFILE' EXIT
```
Comment (DRMacIver) [inline scripts/install.sh]: Same question.

File: scripts/install.sh [hunk 4]
```diff
@@ -23,43 +23,43 @@ env | grep UTF
...
 
 BASE=${BUILD_RUNTIMES-$PWD/.runtimes}
 
-mkdir -p $BASE
+mkdir -p "$BASE"
 
 LOCKFILE="$BASE/.install-lockfile"
 while true; do
-  if mkdir $LOCKFILE 2>/dev/null; then
+  if mkdir "$LOCKFILE" 2>/dev/null; then
```
Comment (DRMacIver) [inline scripts/install.sh]: In which @DRMacIver is made sad by being reminded that this code exists.

File: scripts/basic-test.sh [hunk 5]
```diff
@@ -30,7 +30,7 @@ fi
 
 $PYTEST --runpytest=subprocess tests/pytest
 
-pip install .[datetime]
+pip install ".[datetime]"
```
Comment (DRMacIver) [inline scripts/basic-test.sh]: Not for now, but this reminds me that I think we should probably get rid of our use of extras in tests and start using a requirements.txt for them.

File: scripts/check-ancient-pip.sh (hunk 2 above)
Comment (alexwlchan) [inline scripts/check-ancient-pip.sh]: 1. This is the fix recommended by spellcheck, so I think it’s legit.

2. The variable remains in scope, as this example demonstrates:

    ```console
    $ cat traptest.sh
    #!/usr/bin/env bash
    MESSAGE="hello world"
    trap 'echo $MESSAGE' EXIT
    sleep 2
    exit 1

    $ ./traptest.sh
    hello world
    ```

3. It uses the definition at the point of trap execution, not definition. So you can redefine variables used in the trap:

    ```console
    $ cat traptest.sh
    #!/usr/bin/env bash
    MESSAGE="hello world"
    trap 'echo $MESSAGE' EXIT
    MESSAGE="foo bar"
    sleep 2
    exit 1

    $ ./traptest.sh
    foo bar
    ```

    That doesn’t affect either of our two uses and in practice I think is fine because most of this code is not long for this world, and I can’t be bothered to work out how to stop shellcheck nagging about these lines.

**Comment by alexwlchan** [comment]
> Not quite sure what the best way to do that is though. Maybe just look at all tracked files and check if they have a shebang?

`find . -name '*.sh'`?

**Comment by DRMacIver** [comment]
> find . -name '*.sh'?

The file is ophidian not ophidian.sh (and I'd like to keep it that way)
